node/src/main/scala/coop/rchain/node/balance/TransactionBalances.scala

```diff
@@ -0,0 +1,364 @@
+  ): F[GlobalVaultsInfo] =
+    (1L to targetBlock.body.state.blockNumber).toList.foldLeftM(genesisVaultMap) {
+      case (vaultsInfo, blockNumber) =>
+        for {
+          blocks <- dag.topoSort(blockNumber, Some(blockNumber))
+          resultVaultMap <- blocks.flatten.foldLeftM(vaultsInfo.vaultMaps) {
+                             case (vm, blockHash) =>
+                               for {
+                                 isFinalized <- dag.isFinalized(blockHash)
+                                 result <- if (isFinalized) for {
+                                            block <- blockStore.get(blockHash)
+                                            afterCostDeployMap = block.get.body.deploys
+                                              .foldLeft(vm) {
+                                                case (vaultMaps, pd) =>
+                                                  updateAccountMapFromDeploy(
+                                                    vaultMaps,
+                                                    pd,
+                                                    vaultsInfo.perValidatorVaults.head
+                                                  )
+                                              }
+                                            transactions <- transactionStore.get(
+                                                             Base16.encode(blockHash.toByteArray)
+                                                           )
+                                            res = transactions.get.flatten
+                                              .foldLeft(afterCostDeployMap) {
+                                                case (vaultMap, transaction) =>
+                                                  updateAccountMapFromTransaction(
+                                                    vaultMap,
+                                                    transaction,
+                                                    vaultsInfo.perValidatorVaults
+                                                  )
+                                              }
+                                            g = vaultsInfo.copy(vaultMaps = res)
+                                            result = specialCases.get(
+                                              block.get.body.state.blockNumber.toInt
+                                            ) match {
+                                              case Some(special) => special.handle(g)
+                                              case None          => g
+                                            }
+                                          } yield result.vaultMaps
+                                          else vm.pure[F]
+                               } yield result
+                           }
+          newGlobalVaultInfo = vaultsInfo.copy(vaultMaps = resultVaultMap)
+        } yield newGlobalVaultInfo
+    }
+
+  def getBlockHashByHeight[F[_]: Sync](
```

Comment:
Just a general comment - would be nice to have annotations for functions to easier understand the purpose. And make private functions like this that are not useful in any way as a public API.